### ISSUE TITLE
Inconsistent initial value when using Switch in Form

### ISSUE THREAD

**Issue by xinkule** [ant-design/ant-design#10945]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Version
3.6.2

### Environment
OSX 11, chrome 67.0.3396.87

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/rwk9yw9m7o)

### Steps to reproduce
Declare a Switch component in a separate function component, and use form.getFieldDecorator() to set its initial value.
then when you use form.getFieldValue() to get its value in its father component, you will get undefined.
But, if you just use the Switch directly, not through another encapsulated component, you will get what you want. 
Just see the demo.

### What is expected?
Both ways I could get the correct value I initially set.

### What is actually happening?
The former one I got undefined value.

**Comment by afc163** [comment]
<img width="1077" alt="image" src="https://user-images.githubusercontent.com/507615/41581517-46f66ee8-73d1-11e8-92c9-8542d920b17b.png">

It works well in your codesandbox.

**Comment by xinkule** [comment]
@afc163 nope, what I expected is the text "show check1" appearing, but it didn't, and that's because execute form.getFieldValue("check1") right after <SwitchField /> will get undefined.

**Comment by afc163** [comment]
Check https://github.com/react-component/form/pull/165

**Comment by xinkule** [comment]
@afc163 I checked the link, I don't think we encounter the same problem. My question is why encapsulating Switch component in a function component and using it directly have different behaviors. The former I got "undefined", the latter I got "true".

**Comment by afc163** [comment]
The field value of check1 is inited when first render happens.

**Comment by xinkule** [comment]
@afc163 Then why does `form.getFieldValue("check1") === undefined` ? (I'm not talking about the value of check1 when submiting, I'm saying the value of check1 is undefined right after the statement `<SwitchField form={form} label="check1" name="check1" initialValue={true} />` ) BTW, I'm not sure you really get what I meant

**Comment by igordos** [comment]
@xinkule Look towards `valuePropName`

`<FormItem {...formItemLayout} label='Notification state'>
                            {getFieldDecorator('notificationState', {
                                valuePropName: 'checked',
                                initialValue: notificationState,
                            }
                            )(
                                <Switch
                                    checkedChildren={<Icon type="check" />}
                                    unCheckedChildren={<Icon type="cross" />}
                                />
                            )}
                        </FormItem>`

**Comment by enzoferey** [comment]
As this was a top google result, current way of setting `valuePropName` is described at https://github.com/ant-design/ant-design/issues/20803#issuecomment-601626759.
